### PR TITLE
Threading backgroundfield init

### PR DESCRIPTION
**Pull request by ykempf**
phiprof calls left in there for now, to be investigated also at scale – does it work better now? Posting for CI output too.

### CODE REVIEW

File: backgroundfield/backgroundfield.cpp [hunk 1]
```diff
@@ -41,93 +42,94 @@ void setBackgroundField(
...
+      faceCoord1[0]=1;
+      faceCoord2[0]=2;
+      faceCoord1[1]=0;
+      faceCoord2[1]=2;
+      faceCoord1[2]=0;
+      faceCoord2[2]=1;
+      
+      auto localSize = BgBGrid.getLocalSize();
+      
+      // TODO: These can be threaded now that the dipole field is threadsafe
```
Comment (ursg) [inline backgroundfield/backgroundfield.cpp:62]: I suppose this comment is now moot?

**Review by ursg — APPROVED** [review]
CI output looks good, too.

**Comment by markusbattarbee** [comment]
The only significant diff is from nonthermal moments, which is a small small subpopulation anyway and sensitive to close-to-numerical changes, so I wouldn't be too worried about that.

**Comment by ykempf** [comment]
In this goes?

**Pull request merged**